Scripts/detect_public_api_changes/compare.sh

```diff
@@ -0,0 +1,157 @@
+function setupComparisonRepo() {
+
+    rm -rf $COMPARISON_VERSION_DIR_NAME
+    mkdir $COMPARISON_VERSION_DIR_NAME
+    cd $COMPARISON_VERSION_DIR_NAME
+    git clone -b $BRANCH $REPO
+    cd ..
+}
+
+# Builds the project and temporarily modifies
+# some files to optimize the process
+#
+# $1 - The path to the current project dir
+#
+# Examples
+#
+#  buildProject . # If we're in the current project dir
+#  buildProject ../.. # If we're in the comparison project dir
+function buildProject() {
+
+    # Removing derived data if available
+    rm -rf .build
+
+    # We have to obscure the project file so `xcodebuild` uses the Package.swift to build the module
+    mv Adyen.xcodeproj Adyen.xcode_proj
+    
+    # Copying the Package.swift so we can revert the change done by the next step
+    cp Package.swift Package.sw_ift
+    
+    # Modify the Package.swift file to generate a product/library that contains all targets
+    # so we only have to build it once and can use it to diff all modules
+    $1/Scripts/detect_public_api_changes/package_file_helper.swift Package.swift -add-consolidated-library $ALL_TARGETS_LIBRARY_NAME
+
+    xcodebuild -scheme $ALL_TARGETS_LIBRARY_NAME \
+    -sdk $SDK \
+    -derivedDataPath $DERIVED_DATA_PATH \
+    -destination "platform=iOS,name=Any iOS Device" \
+    -target $TARGET \
+    -quiet \
+    -skipPackagePluginValidation
+    
+    # Reverting the tmp changes
+    rm Package.swift
+    mv Adyen.xcode_proj Adyen.xcodeproj
+    mv Package.sw_ift Package.swift
+}
+
+# Generates an sdk-dump in form of a json file
+#
+# $1 - The module to generate the dump for
+# $2 - The output path for the generated json file
+#
+# Examples
+#
+#  generateSdkDump "AdyenDropIn" "api_dump.json"
+#
+function generateSdkDump() {
+
+    xcrun swift-api-digester -dump-sdk \
+     -module $1 \
+     -o $2 \
+     -I $SDK_DUMP_INPUT_PATH \
+     -sdk $SDK \
+     -target $TARGET
+}
+
+# Compares both versions of the provided module
+# and writes the result into a .md file (Handled by the diff.swift script)
+#
+# $1 - The module to compare
+#
+# Examples
+#
+#  diffModuleVersions "AdyenDropIn"
+#
+function diffModuleVersions() {
+
+    echo "📋 [$1] Generating current sdk dump"
+    generateSdkDump $1 "api_dump.json"
+
+    cd $COMPARISON_VERSION_DIR_NAME/adyen-ios
+
+    echo "📋 [$1] Generating comparison sdk dump"
+    generateSdkDump $1 "../../api_dump_comparison.json"
+
+    cd ../..
+    
+    ./Scripts/detect_public_api_changes/diff.swift "api_dump.json" "api_dump_comparison.json" $1
+    
+    # Cleaning up afterwards
+    rm api_dump.json
+    rm api_dump_comparison.json
+}
+
+# -------------------------------------------------------------------
+# Main Execution
+# -------------------------------------------------------------------
+
+printRunInfo
+
+echo "↘️  Setting up comparison project"
+setupComparisonRepo
+
+# Move into the comparison project
+cd $COMPARISON_VERSION_DIR_NAME/adyen-ios
```

Comment:
Can the hardcoded name be avoided with flattening the dir structure after `git clone` so we don't rely on repo name whatsoever?
I'm just trying to think how we could extract this set of tools into a separate repo/github action so it can be easily reused by other teams.

Reply:
Yes, was also thinking about reusability in the [next version](https://github.com/Adyen/adyen-ios/pull/1718)